drivers/mtd/gd55.c

```diff
@@ -0,0 +1,2065 @@
+                                  (GD55_SECTOR_SHIFT -
+                                   GD55_SECTOR512_SHIFT);
+#else
+              geo->blocksize    = GD55_PAGE_SIZE;
+              geo->erasesize    = GD55_SECTOR_SIZE;
+              geo->neraseblocks = priv->nsectors;
+#endif
+              ret               = OK;
+
+              finfo("blocksize: %" PRId32
+                    " erasesize: %" PRId32
+                    " neraseblocks: %" PRId32 "\n",
+                    geo->blocksize, geo->erasesize, geo->neraseblocks);
+            }
+        }
+        break;
+
+      case BIOC_PARTINFO:
+        {
+          FAR struct partition_info_s *info =
+            (FAR struct partition_info_s *)arg;
+          if (info != NULL)
+            {
+#ifdef CONFIG_MTD_GD55_SECTOR512
+              info->numsectors  = priv->nsectors <<
+                               (GD55_SECTOR_SHIFT - GD55_SECTOR512_SHIFT);
+              info->sectorsize  = GD55_SECTOR512_SIZE;
+#else
+              info->numsectors  = priv->nsectors <<
+                                  (GD55_SECTOR_SHIFT - GD55_PAGE_SHIFT);
+              info->sectorsize  = GD55_PAGE_SIZE;
+#endif
+              info->startsector = 0;
+              info->parent[0]   = '\0';
+              ret               = OK;
+            }
+        }
+        break;
+
+      case MTDIOC_PROTECT:
+        {
+          FAR const struct mtd_protect_s *prot =
+            (FAR const struct mtd_protect_s *)((uintptr_t)arg);
+
+          DEBUGASSERT(prot);
+          ret = gd55_protect(priv, prot->startblock, prot->nblocks);
+        }
+        break;
+
+      case MTDIOC_UNPROTECT:
+        {
+          FAR const struct mtd_protect_s *prot =
+            (FAR const struct mtd_protect_s *)((uintptr_t)arg);
+
+          DEBUGASSERT(prot);
+          ret = gd55_unprotect(priv, prot->startblock, prot->nblocks);
+        }
+        break;
+
+      case MTDIOC_BULKERASE:
+        {
+          /* Erase the entire device */
+
+          gd55_lock(priv);
+          ret = gd55_erase_chip(priv);
+          gd55_unlock(priv);
+        }
+        break;
+
+      case MTDIOC_ERASESTATE:
+        {
+          FAR uint8_t *result = (FAR uint8_t *)arg;
+          *result = GD55_ERASED_STATE;
+
+          ret = OK;
+        }
+        break;
+
+      default:
+        ret = -ENOTTY; /* Bad/unsupported command */
+        break;
+    }
+
+  finfo("return %d\n", ret);
+  return ret;
+}
+
+/****************************************************************************
+ * Name: gd55_readid
+ *
+ * Description:
+ *   Read the device ID.
+ *   - the read ID is stored in the cmdbuf variable of the device structure
+ *
+ * Input Parameters:
+ *   priv       - a reference to the device structure
+ *
+ * Returned Value:
+ *   Success (OK) or fail (negated error code)
+ *
+ ****************************************************************************/
+
+int gd55_readid(FAR struct gd55_dev_s *priv)
```

Comment:
add static

Reply:
Odd that declaration is as static and not picked up by compiler or checkpatch. Thanks.

Reply:
Loads of these - sorry. My excuse is that I know I'd been through sorting these but reverted a commit for other "style" changes done in error and missed these. Doh!

Done.